app/controllers/admin/delivery_zones_controller.rb

```diff
@@ -0,0 +1,49 @@
+class Admin::DeliveryZonesController < ApplicationController
+
+    def new
+        @delivery_zone = DeliveryZone.new    
+    end
+
+    def create
+        @delivery_zone = DeliveryZone.new(delivery_zone_params)
+        if @delivery_zone.save
+            redirect_to admin_delivery_zones_url
+        else
+            render 'new'
+        end
+    end
+
+    def index
+        @delivery_zones = DeliveryZone.all.order(:name)
+    end
+  
+    def show
+        @delivery_zone = DeliveryZone.find(params[:id])
+    end
+
+    def edit
+        @delivery_zone = DeliveryZone.find(params[:id])
+        redirect_to admin_restaurants_url if @delivery_zone.nil?
+    end
+
+    def update
+        @delivery_zone = DeliveryZone.find(params[:id])
+        if @delivery_zone.update_attributes(delivery_zone_params)
+            render 'show'
+        else
+            render 'edit'
+        end
+    end
+
+    def destroy
+        DeliveryZone.find(params[:id]).destroy
```

Comment:
Any reason you didn't use `destroy!`? If this fails, we will not be able to catch it.

Reply:
To be honest, I try not to use ```destroy``` at all. I prefer to set records as disabled with a flag and hide/show them accordingly. Thankfully, I haven't ran into any kind of data privacy/ownership issues. I'd prefer to preserve data as much as possible for historical purposes. Either way, you are correct. There should be better exception handling in these admin controllers.

Reply:
I see. I'm asking only because I often see developers avoiding the 'bang' methods because they don't want to 'break' things - which actually leads to more problems because it makes the expectation/intention of their code unclear and pushes the errors downstream, where it becomes harder to debug.